projects/storefrontlib/src/cms-components/cart/add-to-cart/add-to-cart.component.ts

```diff
@@ -43,16 +55,41 @@ export class AddToCartComponent implements OnInit, OnDestroy {
     quantity: new FormControl(1, { updateOn: 'blur' }),
   });
 
+  constructor(
+    modalService: ModalService,
+    currentProductService: CurrentProductService,
+    cd: ChangeDetectorRef,
+    activeCartService: ActiveCartService,
+    componentData: CmsComponentData<model>
+  );
+
+  /**
+   * @deprecated since 3.4
+   */
+  constructor(
+    modalService: ModalService,
+    currentProductService: CurrentProductService,
+    cd: ChangeDetectorRef,
+    activeCartService: ActiveCartService
+  );
+
   constructor(
     protected modalService: ModalService,
     protected currentProductService: CurrentProductService,
-    protected cd: ChangeDetectorRef,
-    protected activeCartService: ActiveCartService
+    private cd: ChangeDetectorRef,
+    protected activeCartService: ActiveCartService,
+    @Optional() protected componentData?: CmsComponentData<model>
   ) {}
 
   ngOnInit() {
+    this.componentData?.data$.subscribe((data: model) => {
+      if (data.inventoryDisplay && data.inventoryDisplay == 'true') {
```

Comment:
similar to https://github.com/SAP/spartacus/pull/11978#discussion_r617451306

Reply:
```suggestion
      if Boolean(data.inventoryDisplay) {
```

**OR turn the variable into an observable**

```suggestion
     showInventoryDisplay$ = this.component.data$.pipe(filter(Boolean),map((data) => data.inventoryDisplay } ))
```